### PR TITLE
Add service monitor

### LINKED ISSUE #5: Add ServiceMonitor
Add a ServiceMonitor template for fetching metrics from the telemetry endpoint of Dex.

### PR DESCRIPTION
**Pull request by mkoertgen**
#### Overview

Add a Prometheus ServiceMonitor template and values.

#### What this PR does / why we need it

Closes #5 

#### Checklist

- [x] Change log updated in `Chart.yaml` (see the contributing guide for details)
- [x] Chart version bumped in `Chart.yaml` (see the contributing guide for details)
- [x] Documentation regenerated by running `make docs`

### CODE REVIEW

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:1]: nit

```suggestion
{{- if and .Values.serviceMonitor.enabled }}
```

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:9]: This is the default

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:6]: Any reason why the namespace needs to be overridden?

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:13]: This should already be included in `dex.labels` template:

https://github.com/dexidp/helm-charts/blob/master/charts/dex/templates/_helpers.tpl#L49

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:12]: Is this an officially supported feature? Can you please link to the documentation?

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:17]: Is there a practical use case for adding custom labels to service monitors?

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:30]: nit: indentation

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:25]: Isn't this only necessary when the service monitor is in a different namespace?

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:28]: Isn't /metrics the default?

File: charts/dex/values.yaml
Comment (sagikazarmark) [inline charts/dex/values.yaml:212]: This won't show up in the docs.

File: charts/dex/values.yaml
Comment (sagikazarmark) [inline charts/dex/values.yaml:214]:
```suggestion
  # -- Enable Prometheus ServiceMonitor
```

Maybe add the link to service monitor here.

File: charts/dex/values.yaml
Comment (sagikazarmark) [inline charts/dex/values.yaml:215]:
```suggestion
  enabled: false

```

Add trailing lines between options with documentation.

File: charts/dex/values.yaml
Comment (sagikazarmark) [inline charts/dex/values.yaml:213]:
```suggestion
serviceMonitor:
```

**Review by sagikazarmark — CHANGES_REQUESTED** [review]

File: charts/dex/templates/servicemonitor.yaml
Comment (mkoertgen) [reply charts/dex/templates/servicemonitor.yaml:6]: Depends on the prometheus-operator configuration, i.e. where it looks for serviceMonitors. Sometimes you want to put the serviceMonitor into the monitoring-namespace in case the operator only looks there. So it should be possible to override the namespace with the release-namespace as the default.

File: charts/dex/templates/servicemonitor.yaml
Comment (mkoertgen) [reply charts/dex/templates/servicemonitor.yaml:28]: Right. Could probably be omitted.

**Comment by mkoertgen** [comment]
Hi @sagikazarmark, thanks for reviewing. Will update shortly.

File: charts/dex/templates/servicemonitor.yaml
Comment (mkoertgen) [reply charts/dex/templates/servicemonitor.yaml:12]: No, it's not. I will remove it.

File: charts/dex/templates/servicemonitor.yaml
Comment (mkoertgen) [reply charts/dex/templates/servicemonitor.yaml:13]: Yes, it is. Will remove it.

File: charts/dex/templates/servicemonitor.yaml
Comment (mkoertgen) [reply charts/dex/templates/servicemonitor.yaml:9]: Yes, it is. Will document it.

File: charts/dex/templates/servicemonitor.yaml
Comment (mkoertgen) [reply charts/dex/templates/servicemonitor.yaml:28]: No, no default that I am aware of. Docs: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#endpoint

**Comment by dirien** [comment]
Hi @mkoertgen,

Is there smth missing in this PR? We are eagerly waiting for this 😄 Thx.

**Comment by mkoertgen** [comment]
@dirien Not 100% sure. I updated outdated code changes. 

If I understand correctly the check-workflow needs an approval to run. 
@sagikazarmark Would you mind?

**Comment by dirien** [comment]
The DCO is on your side @mkoertgen you can `git commit --amend -s` and  git push the branch. Worst case, would be that you either need to squash your branch to a single commit or have to sign each commit. I prefer to squash.

So lets see what the maintainer will say!

Thanks for your effort, btw!

**Comment by mkoertgen** [comment]
Thx for the clarification, @dirien. Merged master, squashed & signed-off.

**Comment by dirien** [comment]
I have another PR open in dex but I have the feeling, either the exising member are too busy to maintain the project in full extend or the upstream is slowly dying :(

Would volunteer to join the organization to help

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:10]:
```suggestion
{{- with .Values.serviceMonitor.namespace }}
  namespace: {{ . }}
{{- end }}
```

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:15]:
```suggestion
    {{- with .Values.serviceMonitor.labels }}
      {{ toYaml . | indent 4}}
    {{- end }}
```

File: charts/dex/templates/servicemonitor.yaml
Comment (sagikazarmark) [inline charts/dex/templates/servicemonitor.yaml:26]: Isn't this the default? Can we leave it out if it is?

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Please run make docs and bumb the chart version. See details in the contributing guide.

**Comment by dirien** [comment]
hi @mkoertgen, did you had time to check the suggestion of @sagikazarmark?

**Comment by mkoertgen** [comment]
Hi @dirien ,
No sorry, I have been busy over the week. Will update the PR this afternoon.

**Comment by mkoertgen** [comment]
Hi @sagikazarmark ,
Can you point me to the contributing guide? Cannot find it.
Also I don't use "make" on a daily basis but I am familiar with helm-docs.

Bumping the version would be here?
- https://github.com/dexidp/helm-charts/blob/83db6aec1634eea20e5850fda204ac4729fcedb9/charts/dex/Chart.yaml#L4

What version would you like it to be? Like 0.6.5 -> 0.7.0?

Look, I love to contribute but I feel stupid and lost here. 
Would you mind taking over the PR and modify or close altogether at your convenience?

**Comment by dirien** [comment]
@mkoertgen I would say 0.7.0 as introducing a new resource is not a patch for me and justifies the minor version bump. But thats IMHO only.

**Comment by sagikazarmark** [comment]
@mkoertgen please see #73 and test it in your environment if possible.

**Pull request closed without merging**
